test/ui-test/extensionUITest.ts

```diff
@@ -0,0 +1,64 @@
+import { expect } from 'chai';
+import { ActivityBar, ExtensionsViewSection, ExtensionsViewItem, VSBrowser, NotificationType } from 'vscode-extension-tester';
+import { KNativeConstants } from './common/constants';
+import { getNotifications } from './common/testUtils';
+/**
+ * @author Ondrej Dockal <odockal@redhat.com>
+ */
+export function extensionsUITest(): void {
+  describe('Knative extension', () => {
+    it('should be installed among extensions view', async function context() {
+      this.timeout(5000);
+      const view = new ActivityBar().getViewControl('Extensions');
+      const sideBar = await view.openView();
+      const section = (await sideBar.getContent().getSection('Installed')) as ExtensionsViewSection;
+      const item = await section.findItem(`@installed ${KNativeConstants.KNATIVE_EXTENSION_NAME}`);
+      expect(item).to.be.an.instanceOf(ExtensionsViewItem);
+      expect(await item.getTitle()).to.equal(KNativeConstants.KNATIVE_EXTENSION_NAME);
+    });
+
+    it('should have Knative Activity Bar object available', async function context() {
+      this.timeout(5000);
+      const view = new ActivityBar().getViewControl(KNativeConstants.KNATIVE_EXTENSION_NAME);
+      const sideBar = await view.openView();
+      // check that no notification error appeared
+      VSBrowser.instance.driver.wait(async function receiveNotifications() {
+        const notifs = await getNotifications(NotificationType.Error, NotificationType.Warning);
+        return notifs.length === 0;
+      }, 3000);
+      const titlePart = sideBar.getTitlePart();
+      expect(await titlePart.getTitle()).to.equal(KNativeConstants.KNATIVE_EXTENSION_BAR_NAME);
+    });
+
+    it('should be ready for usage, requires oc on the path to be logged into cluster', async function context() {
```

Comment:
You are right, this is not precise. I was assuming that oc is one of the requirements, but it could be kubectl as well. I will change it to something like "requires to be logged into cluster" right?